uPortal-api/uPortal-api-rest/src/main/java/org/apereo/portal/rest/LayoutRESTController.java

```diff
@@ -115,65 +112,31 @@ public ModelAndView getRESTController(
 
             final IUserProfile profile = upm.getUserProfile();
             final DistributedUserLayout userLayout = userLayoutStore.getUserLayout(person, profile);
-            Document document = userLayout.getLayout();
-
-            NodeList portletNodes = null;
-            if (tab != null) {
-                NodeList folders = document.getElementsByTagName("folder");
-                for (int i = 0; i < folders.getLength(); i++) {
-                    Node node = folders.item(i);
-                    if (tab.equalsIgnoreCase(
-                            node.getAttributes().getNamedItem("name").getNodeValue())) {
-                        TabListOfNodes tabNodes = new TabListOfNodes();
-                        tabNodes.addAllChannels(node.getChildNodes());
-                        portletNodes = tabNodes;
-                        break;
-                    }
-                }
-            } else {
-                portletNodes = document.getElementsByTagName("channel");
-            }
+            final Document document = userLayout.getLayout();
+
+            final NodeList portletNodes = getNodeList(tab, document);
+
```

Comment:
It may be better to refactor the code in another commit ?

Reply:
commit or PR?  I could see a different commit for a cleaner review.  If the code refactor is associated with the code changes, I'd stick with the same PR.

Reply:
a commit is sufficient as in history, after merged, it won't be linked to the feature.